applications/Chat/evaluate/metrics.py

```diff
@@ -136,34 +206,128 @@ def calculate_precision_recall_f1(preds: list, targets: list) -> dict:
     return precision_recall_f1
 
 
-def precision(preds: list, targets: list) -> dict:
+def precision(preds: List[str], targets: List[str], language: str) -> Dict[str, float]:
     """Calculate Precision Metric
     (design for classification and extraction categories)
```

Comment:
(Designed for classification and extraction categories)

Reply:
I will remove this.